File: ukol_03/Java/3DSouradnice/src/molecule/Atom.java
package molecule;

import graph.Node;

import java.util.Map;

/**
 * Created by Martin Šícho on 7.11.14.
 */
public class Atom extends Node {
    private double x;
    private double y;
    private double z;

    // constructors

    public Atom(double x, double y, double z) {
        super();
        setCoords(x, y, z);
    }

    public Atom(String name, double x, double y, double z) {
        super(name);
        setCoords(x, y, z);
    }

    public Atom(double[] coords) {
        super();
        if (coords.length < 3) {
            throw new ArrayStoreException(String.format("Array size is insufficient (%d).", coords.length));
        }
        setCoords(coords[0], coords[1], coords[2]);
    }

    public Atom(String name, double[] coords) {
        super(name);
        if (coords.length < 3) {
            throw new ArrayStoreException(String.format("Array size is insufficient (%d).", coords.length));
        }
        setCoords(coords[0], coords[1], coords[2]);
    }

    // non-static methods

    public void getCoords(double[] coords) {
        if (coords.length < 3) {
            throw new ArrayStoreException(String.format("Array size is insufficient (%d).", coords.length));
        }
        coords[0] = x;
        coords[1] = y;
        coords[2] = z;
    }

    public void getCoords(Map<Character, Double> coords_map) {
        coords_map.put('x', x);
        coords_map.put('y', y);
        coords_map.put('z', z);
    }

    public void setCoords(double x, double y, double z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }

    // overrides

    @Override
    public String toString() {
        return String.format("molecule.Atom[order:%d,name:%s]", getNodeCount(), getName());
    }


    // getters & setters

    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }

    public double getZ() {
        return z;
    }

    public void setZ(double z) {
        this.z = z;
    }
}

File: ukol_03/Python/3DSouradnice/molecule/Atom.py
import Node


class Atom(Node.Node):
    _x = int()
    _y = int()
    _z = int()

    def __init__(self, x, y, z, name=""):
        Node.Node.__init__(self, name=name)
        self._x = x
        self._y = y
        self._z = z

    def __repr__(self):
        return "Atom[order:{0},name:{1}]".format(self.getNodeCount(), self.getName())

    def setCoords(self, x, y, z):
        self._x = x
        self._y = y
        self._z = z

    def getX(self):
        return self._x

    def getY(self):
        return self._y

    def getZ(self):
        return self._z

    def setX(self, x):
        self._x = x

    def setY(self, y):
        self._y = y

    def setZ(self, z):
        self._z = z